src/fs.rs

```diff
@@ -151,6 +155,141 @@ pub fn create_workspace_archive(working_root_dir: &str, execution_id: &str) -> R
     }
 }
 
+pub fn create_yaml_backup_file<P>(
+    working_root_dir: P,
+    chart_name: String,
+    resource_name: Option<String>,
+    content: String,
+) -> Result<String, std::io::Error>
```

Comment:
OPT: Maybe return a CommandError here, so this function takes care of sending back safe and unsafe message properly.